libs/core/langchain_core/retrievers.py

```diff
@@ -190,15 +190,40 @@ def invoke(
 
             retriever.invoke("query")
         """
+        from langchain_core.callbacks.manager import CallbackManager
```

Comment:
```suggestion
        from langchain_core.callbacks import CallbackManager
```
if importing from init is a performance problem, probably worth implementing `getattr` performance thing in callbacks/init, and adding comment here that import is for load perf

if not performance, move import to top of file